phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixConnection.java

```diff
@@ -128,12 +129,12 @@
     private PMetaData metaData;
     private final PName tenantId;
     private final String datePattern;
-    
+    private int statementExecutionsCount;
```

Comment:
Yeah, possibly. I'll add more tests around UPSERT SELECT to make sure cloned connections carry state correctly.